aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -1999,7 +1999,7 @@ def register_resolver(func: Callable):
 
         return register_resolver
 
-    def resolve(self, event, context) -> dict[str, Any]:
+    def resolve(self, event: dict[str, Any], context: LambdaContext) -> dict[str, Any]:
```

Comment:
Hi @rafrafek! Thank you so much for submitting this PR to improve the type annotations.

I agree with this PR and it is a good opportunity to refactor our examples in the EventHandler utility to add a better type annotation in the `lambda_handler` function signature. Let me do this and then we can merge.

Reply:
Oh no, if I do that I'll need to refactor the code highlight and we have a PR for that: https://github.com/aws-powertools/powertools-lambda-python/pull/4959.

Approving.

Reply:
Hi @leandrodamascena, thank you for reviewing the PR and for your approval!